ServiceMonitoringFunction/tst/com.amazon.vas.serviceordermonitoringlambda/util/TestDataBuilder.java

```diff
@@ -0,0 +1,185 @@
+                .merchantId(null).serviceCategory(null)
+                .groupingCriteria(buildGroupingCriteria_cityAndSlotStartTime()).build();
+        return getJobMetricsInput;
+    }
+
+    public GetJobMetricsInputBO buildGetJobMetricsInputBO_withMerchantIdAndSlotStartTime(){
+        GetJobMetricsInputBO getJobMetricsInputBO = GetJobMetricsInputBO.builder().city(TEST_CITY)
+                .merchantId(null).serviceCategory(null)
+                .groupingCriteria(buildGroupingCriteria_merchantIdAndSlotStartTime()).build();
+        return getJobMetricsInputBO;
+    }
+
+    private List<String> buildGroupingCriteria_cityAndSlotStartTime() {
+        return new ArrayList<>(Arrays.asList(CITY, SLOT_START_TIME));
+    }
+    private List<String> buildGroupingCriteria_merchantIdAndSlotStartTime() {
+        return new ArrayList<>(Arrays.asList(MERCHANT_ID, SLOT_START_TIME));
+    }
+
+    public ArrayList<JobDetailsBO> buildJobDetailsList() {
+        final Gson gson = new Gson();
+        final List<String> jobDetailsJson = buildJobDetailsJson();
+        final ArrayList<JobDetailsBO> expectedJobDetailsList = new ArrayList<>();
+        for (String job : jobDetailsJson) {
+            expectedJobDetailsList.add(gson.fromJson(job, JobDetailsBO.class));
+        }
+        return expectedJobDetailsList;
+    }
+
+    public GetJobMetricsOutput buildGetJobMetricsOutput_withCityAndSlotStartTime(GetJobMetricsOutputBO getJobMetricsOutputBO) {
+        return GetJobMetricsOutput.builder().metaData(getJobMetricsOutputBO.getMetaData())
+                .jobMetricsMap(getJobMetricsOutputBO.getJobMetricsMap()).build();
+    }
+
+
+    public GetJobMetricsOutputBO buildGetJobMetricsOutputBO_withCityAndSlotStartTime() {
+
+        final Map<String, Map<String, String>> metaData = new HashMap<>();
+        final Map<List<String>, JobAggregatedMetricsBO> jobMetricsMap = new HashMap<>();
+        for(int i=0;i<10;i++){
+            String tempCity = "City-"+i;
+            metaData.put(tempCity, new HashMap<String, String>() {{
+                put("name", tempCity);
+                put("ID", tempCity);
+            }});
+        }
+
+        for(int i=0;i<numberOfTestAggregatedJobs;i++){
+            int jobsCount = randomNumberOfJobs.get(i);
+            final JobAggregatedMetricsBO jobAggregatedMetricsBO = JobAggregatedMetricsBO.builder()
+                    .totalJobsCount(jobsCount)
+                    .statusNotUpdatedCount(jobsCount)
+                    .otaFailureCount(jobsCount)
+                    .etaDelayCount(jobsCount).build();
+            final List<String> groupingCriteriaValues = buildGroupingCriteriaValues_cityAndSlotStartTime(i);
+            jobMetricsMap.put(groupingCriteriaValues, jobAggregatedMetricsBO);
+        }
+
+        return GetJobMetricsOutputBO.builder().metaData(metaData).jobMetricsMap(jobMetricsMap).build();
+    }
+
+    public GetJobMetricsOutputBO buildGetJobMetricsOutputBO_withMerchantIdAndSlotStartTime() {
```

Comment:
These 2 functions are very similar. can you just create single function and.pass the Filter keys as argument ?